src/filter/filter.c

```diff
@@ -22,28 +22,25 @@
 
 #include "filter.h"
 
-//Implements a fast Exponential Moving Average IIR filter
-
-//This macros defines an alpha value between 0 and 1
-#define DSP_EMA_I32_ALPHA(x) ( (unsigned short)(x * 65535) )
-
-static int dsp_ema_i32(int in, int average, unsigned short alpha)
-{
-    long long tmp0; //calcs must be done in 64-bit math to avoid overflow
-    tmp0 = (long long)in * (alpha) + (long long)average * (65536 - alpha);
-    return (int)((tmp0 + 32768) / 65536); //scale back to 32-bit (with rounding)
-}
-
+//Implements a fast Exponential Moving Average filter
+#define MIN_ALPHA 0.0001
 void init_filter(Filter *filter, float alpha)
 {
-    filter->alpha = alpha;
-    filter->current_value = 0;
+        filter->max_samples = 1 / (alpha != 0 ? alpha : MIN_ALPHA);
```

Comment:
This needs to change as it allows for an alpha value below the MIN_ALPHA value
```c
filter->max_samples = 1 / max(alpha, MIN_ALPHA);
```

Reply:
Also... this is a float being assigned to an int type ja?  I assume there is an implicit cast in there but not sure when it happens.  Double check it works as you believe it should.

Reply:
Ok, addressing both points.